### PR TITLE
[ELY-134] Javadoc adding

### PR DESCRIPTION
**Pull request by honza889**
~Based on https://github.com/wildfly-security/wildfly-elytron/pull/964 (necessary to generate API-only javadoc)~

* rework of audit javadoc
* fixes of invalid {@link}s with generic types params
* adding javadoc where missing and included in api

https://issues.jboss.org/browse/ELY-134

### CODE REVIEW

**Comment by darranl** [comment]
I don't think this really needs to depend on the other PR, can you rebase this so we just have one commit - the two can be merged independently.

**Comment by honza889** [comment]
@darranl updated

File: src/main/java/org/wildfly/security/audit/SyslogAuditEndpoint.java [hunk 1]
```diff
@@ -33,7 +33,7 @@
 import javax.net.SocketFactory;
 
 /**
- * An {@link AuditEndpoint} that logs to syslog.
+ * An audit endpoint that logs to syslog server.
```
Comment (darranl) [inline src/main/java/org/wildfly/security/audit/SyslogAuditEndpoint.java]: Not going to hold up the PR for this.

For changes like this is there a reason to remove all links?  These allow automatic cross referencing in the generated javadoc.

Comment (honza889) [reply src/main/java/org/wildfly/security/audit/SyslogAuditEndpoint.java]: This links only duplicates links generated from "implements" of class. So based on recommendation:
> Only use @link on the first reference to a specific class or method. Use @code for subsequent references. This avoids excessive hyperlinks cluttering up the Javadoc.

**Pull request merged**